source/retarget.cpp

```diff
@@ -152,7 +157,15 @@ extern "C" FILEHANDLE PREFIX(_open)(const char* name, int openmode) {
         init_serial();
         return 2;
     }
-    #endif
+#ifdef YOTTA_CFG_DEBUG_OPTIONS_COVERAGE
+    else if(coverage_report) {
+        init_serial();
+        notify_coverage_start(name);
+        return 3;
+    }
+#endif
+
+#endif
```

Comment:
... and all the other places in the file that make a reference to the offset `3` in the file handle table.